content/foundation/glossary.md

```diff
@@ -368,6 +396,18 @@ process](voting.html).
 
 </dd>
 
+<dt>Mentor  {#Mentor}</dt>
+
+<dd>
+
+Typically, it refers to the mentors of [podlings](#Podling).
```

Comment:
I'd say "also called 'incubation mentors'", that's a way to clarify the type of mentorship